File: scripts/fix-html.js
const fs = require('fs');
const path = require('path');
const { glob } = require('glob');

function unescapeHtml(content) {
    return content
        .replace(/&amp;/g, '&')
        .replace(/&lt;/g, '<')
        .replace(/&gt;/g, '>')
        .replace(/&quot;/g, '"')
        .replace(/&#39;/g, "'");
}

function fixHtmlFiles() {
    try {
        const files = glob.sync('**/*.html', { ignore: ['node_modules/**', 'scripts/**'] });
        
        for (const file of files) {
            console.log(`Processing ${file}...`);
            let content = fs.readFileSync(file, 'utf8');
            
            // First unescape any double-escaped content
            content = unescapeHtml(content);
            
            // Fix self-closing tags
            content = content.replace(/<(meta|link|hr|img|input|br)([^>]*?)(?<!\/)\s*>/g, '<$1$2 />');
            
            // Fix ID values to use dashes
            content = content.replace(/id="([^"]+)"/g, (match, id) => {
                const newId = id.toLowerCase().replace(/([a-z])([A-Z])/g, '$1-$2');
                return `id="${newId}"`;
            });
            
            // Fix class values to use dashes
            content = content.replace(/class="([^"]+)"/g, (match, classes) => {
                const newClasses = classes.split(' ')
                    .map(cls => cls.toLowerCase().replace(/([a-z])([A-Z])/g, '$1-$2'))
                    .join(' ');
                return `class="${newClasses}"`;
            });
            
            fs.writeFileSync(file, content, 'utf8');
            console.log(`Fixed ${file}`);
        }
        
        console.log('All HTML files have been fixed');
    } catch (error) {
        console.error('Error:', error);
    }
}

fixHtmlFiles();

File: scripts/fix-html.py
import os
import re
from glob import glob

def fix_html_files():
    try:
        files = glob('**/*.html', recursive=True)
        
        for file in files:
            print(f'Processing {file}...')
            with open(file, 'r', encoding='utf-8') as f:
                content = f.read()
            
            # Fix self-closing tags
            content = re.sub(r'<(meta|link|hr|img|input|br)[^>]*[^/]>', 
                           lambda m: m.group().replace('>', ' />'), 
                           content)
            
            with open(file, 'w', encoding='utf-8') as f:
                f.write(content)
            print(f'Fixed {file}')
        
        print('All HTML files have been fixed')
    except Exception as e:
        print('Error:', e)

if __name__ == '__main__':
    fix_html_files() 